GitDepend/Visitors/RemoveDependencyVisitor.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitDepend.Busi;
+using GitDepend.Configuration;
+
+namespace GitDepend.Visitors
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class RemoveDependencyVisitor : IVisitor
+    {
+        private readonly IGitDependFileFactory _factory;
+        private readonly string _dependencyNameToRemove;
+        private string _foundDependencyDirectory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoveDependencyVisitor"/> class.
+        /// </summary>
+        public RemoveDependencyVisitor(string dependencyNameToRemove)
+        {
+            _factory = DependencyInjection.Resolve<IGitDependFileFactory>();
+            _dependencyNameToRemove = dependencyNameToRemove;
+            _foundDependencyDirectory = string.Empty;
+        }
+
+        /// <summary>
+        /// The return code
+        /// </summary>
+        public ReturnCode ReturnCode { get; set; }
+
+        /// <summary>
+        /// Visits a project dependency.
+        /// </summary>
+        /// <param name="directory">The directory of the project.</param>
+        /// <param name="dependency">The <see cref="Dependency" /> to visit.</param>
+        /// <returns>
+        /// The return code.
+        /// </returns>
+        public ReturnCode VisitDependency(string directory, Dependency dependency)
+        {
+            string dir;
+            ReturnCode returnCode;
+
+            //visit dependency and get the configuration
+            var configFile = _factory.LoadFromDirectory(directory, out dir, out returnCode);
+            if (returnCode == ReturnCode.Success)
+            {
+                if (configFile.Name == _dependencyNameToRemove)
+                {
+                    _foundDependencyDirectory = dir;
```

Comment:
This seems wrong. `dir` is the directory where the projects `GitDepend.json` file was located. NOT the directory where the dependencies `GitDepend.json` file is. If you are trying to load the config for the dependency you need to use the given `directory` as the base, and the directory specified in the `Dependency` to calculate the dependency directory.